### DISCUSSION TITLE
Optionally expect heartbeats from the server

### DISCUSSION THREAD

**Discussion by majek** [sockjs/sockjs-client#560] [Ideas]
The server may send an option telling the client how often it will send 'heartbeat' frames.

We should read this option and, indeed, expect heartbeats. We should kill the session on a heartbeat miss.

**Comment by glasser** [comment]
There was a lot of activity on this issue but it's still open. Does the feature work (and it should be closed)?

**Comment by hyperthunk** [comment]
@glasser I'm not 100% sure, but I'll discuss with @majek when I get a chance and either merge or close it.

**Comment by mokesmokes** [comment]
Any update?

**Comment by triopter** [comment]
It looks like all of this was on a "dev" branch that never got merged.  Is there any plan to merge this into master and release it?

**Comment by majek** [comment]
The fix was way harder then anticipated and required major changes to the codebase, and eventually I gave up. Sorry.

**Comment by 3rd-Eden** [comment]
If you use [primus](https://github/primus/primus) on top of sockjs it will send these heartbeat packages for you. From client -> server and the server will automatically disconnect if a heartbeat missing/timed out. See https://github.com/primus/primus#heartbeats-and-latency

So if you really, this might be a viable option.

**Comment by triopter** [comment]
@majek that's too bad!  We're testing out the "dev" branch with the heartbeats right now (it seems to be up to date w/ v0.3.4) and it appears to be working.

@3rd-Eden Thanks for the pointer to Primus -- we'll definitely take a look at that!

**Comment by rikoe** [comment]
Hi. This issue has been open for almost 5 years now. It was recently discussed in #297. Is there any plan to implement this feature? The client not being aware of the server disappearing from the network seems to be a major flaw.

What is the "dev" branch mentioned above that has never been merged if one wants to take a look at it?

**Comment by harshakakumanu** [comment]
Are there any plans in implementing this issue? Is there any work around ?
